services-geocoding/src/main/java/com/mapbox/api/geocoding/v6/MapboxV6BaseGeocoding.java

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.api.geocoding.v6;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.mapbox.core.MapboxService;
+
+/**
+ * Base class for entry points to Mapbox V6 geocoding: forward geocoding, reverse geocoding
+ * and batch geocoding. See derived classes for more information.
+ * @param <T> response type.
+ */
+public abstract class MapboxV6BaseGeocoding<T> extends MapboxService<T, V6GeocodingService> {
+
+  @NonNull
+  protected abstract String accessToken();
```

Comment:
These fields are supposed to be set by a user via Builder class which is public. We use these fields later in sub classes when we make http request.

I think that in general we should keep everything non-public until proven otherwise, so they're are with minimal possible visibility level.

Reply:
I see your point and it makes sense.
However, I see multiple problems with this approach:
1. Inconsistency: user created the object but can't "see" it: how will they test it, for example? They'd want something like: "when I receive this input data I create these options".
2. User can't modify the existing object based on the actual data. You can imagine creating an interceptoe that would, for example, check if `clientAppName` is null and, if it is, set it to some value.